src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -353,12 +354,36 @@ private BoundExpression CreateTupleLiteralConversion(SyntaxNode syntax, BoundTup
             if (targetType.IsTupleType)
             {
                 var destTupleType = (TupleTypeSymbol)targetType;
-                // do not lose the original element names in the literal if different from names in the target
 
                 TupleTypeSymbol.ReportNamesMismatchesIfAny(targetType, sourceTuple, diagnostics);
 
                 // Come back to this, what about locations? (https://github.com/dotnet/roslyn/issues/11013)
```

Comment:
I think so.

---
In reply to: [97434805](https://github.com/dotnet/roslyn/pull/16659#discussion_r97434805) [](ancestors = 97434805)